go/vt/topo/keyspace.go

```diff
@@ -270,26 +273,77 @@ func (ts *Server) UpdateKeyspace(ctx context.Context, ki *KeyspaceInfo) error {
 	return nil
 }
 
-// FindAllShardsInKeyspace reads and returns all the existing shards in
-// a keyspace. It doesn't take any lock.
-func (ts *Server) FindAllShardsInKeyspace(ctx context.Context, keyspace string) (map[string]*ShardInfo, error) {
+// FindAllShardsInKeyspaceConfig controls the behavior of
+// Server.FindAllShardsInKeyspace.
+type FindAllShardsInKeyspaceConfig struct {
+	// Concurrency controls the maximum number of concurrent calls to GetShard.
+	// If unspecified, Concurrency is set to 1.
+	Concurrency int
+}
+
+// FindAllShardsInKeyspace reads and returns all the existing shards in a
+// keyspace. It doesn't take any lock.
+//
+// If cfg is non-nil, it is used to configure the method's behavior. Otherwise,
+// a default configuration is used.
+func (ts *Server) FindAllShardsInKeyspace(ctx context.Context, keyspace string, cfg *FindAllShardsInKeyspaceConfig) (map[string]*ShardInfo, error) {
+	// Apply any necessary defaults.
+	if cfg == nil {
+		cfg = &FindAllShardsInKeyspaceConfig{}
+	}
+	if cfg.Concurrency == 0 {
```

Comment:
since this is `int`, we need to check `if cfg.Concurrency < 1` or else we may panic later

Reply:
(also can you add a unit test for this?)

Reply:
You bet!